lib/src/universal_ble_platform_interface.dart

```diff
@@ -71,11 +71,29 @@ abstract class UniversalBlePlatform {
   }
 
   void updateScanResult(BleDevice bleDevice) {
-    onScanResult?.call(bleDevice);
+    try {
+      final onScanResult = this.onScanResult;
+      if (onScanResult != null) {
+        onScanResult(bleDevice);
+      }
+    } catch (e) {
+      // report error for user to see
+      debugPrint("Error in UniversalBle.onScanResult");
+      debugPrint(e.toString());
+    }
   }
 
   void updateConnection(String deviceId, bool isConnected, [String? error]) {
-    onConnectionChange?.call(deviceId, isConnected, error);
+    try {
+      final onConnectionChange = this.onConnectionChange;
+      if (onConnectionChange != null) {
+        onConnectionChange(deviceId, isConnected, error);
+      }
+    } catch (e) {
+      // report error for user to see
+      debugPrint("Error in UniversalBle.onConnectionChange");
+      debugPrint(e.toString());
+    }
     _connectionStreamController?.add((
       deviceId: deviceId,
       isConnected: isConnected,
```

Comment:
The real issue for #92 was coming from here. I had some `onConnectionChange` listener which internally threw because of some issue on my side that popped up when updating universal_ble from 0.12.0 to 0.14.0.

Because of that failure, `_connectionStreamController?.add` was never called, causing the `UniversalBle.connect` to hang until timeouot.

Reply:
CC @rohitsangwan01